pkg/daemon/ovs.go

```diff
@@ -495,11 +499,82 @@ func providerBridgeExists() (bool, error) {
 }
 
 // Add host nic to br-provider
-// A physical Ethernet device that is part of an Open vSwitch bridge should not have an IP address. If one does, then that IP address will not be fully functional.
-// More info refer http://docs.openvswitch.org/en/latest/faq/issues
+// MAC, MTU, IP addresses & routes will be transferred to br-provider
 func configProviderNic(nicName string) error {
-	_, err := ovs.Exec(ovs.MayExist, "add-port", "br-provider", nicName)
-	return err
+	brName := util.UnderlayBridge
+	nic, err := netlink.LinkByName(nicName)
+	if err != nil {
+		return fmt.Errorf("failed to get nic by name %s: %v", nicName, err)
+	}
+	bridge, err := netlink.LinkByName(brName)
+	if err != nil {
+		return fmt.Errorf("failed to get bridge by name %s: %v", brName, err)
+	}
+
+	sysctlDisableIPv6 := fmt.Sprintf("net.ipv6.conf.%s.disable_ipv6", brName)
+	disableIPv6, err := sysctl.Sysctl(sysctlDisableIPv6)
+	if err != nil {
+		return fmt.Errorf("failed to get sysctl %s: %v", sysctlDisableIPv6, err)
+	}
+	if disableIPv6 != "0" {
+		if _, err = sysctl.Sysctl(sysctlDisableIPv6, "0"); err != nil {
+			return fmt.Errorf("failed to enable ipv6 on OVS bridge %s: %v", brName, err)
+		}
+	}
+
+	addrs, err := netlink.AddrList(nic, unix.AF_UNSPEC)
```

Comment:
addrs, err := netlink.AddrList(link, netlink.FAMILY_ALL)
FAMILY_ALL is easier to read I think

Reply:
> addrs, err := netlink.AddrList(link, netlink.FAMILY_ALL)
> FAMILY_ALL is easier to read I think

fixed